src/main/java/org/prebid/server/cache/CacheService.java

```diff
@@ -125,21 +128,23 @@ public String getCachedAssetURLTemplate() {
             return Future.succeededFuture(BidCacheResponse.of(Collections.emptyList()));
         }
 
+        final long startTime = clock.millis();
         final long remainingTimeout = timeout.remaining();
         if (remainingTimeout <= 0) {
-            return failResponse(new TimeoutException("Timeout has been exceeded"));
+            return failResponse(new TimeoutException("Timeout has been exceeded"), startTime);
```

Comment:
Actually, this should be processed separately without affecting cache metrics since it is not `Prebid Cache` issue.